ql/termstructures/yield/fittedbonddiscountcurve.hpp

```diff
@@ -197,17 +201,20 @@ namespace QuantLib {
         Real minimumCostValue() const;
         //! clone of the current object
         virtual std::auto_ptr<FittingMethod> clone() const = 0;
-		//! return whether there is a constraint at zero
-		bool constrainAtZero() const;
-		//! return weights being used
-		Array weights() const;
-		//! return optimization method being used
-		boost::shared_ptr<OptimizationMethod> optimizationMethod() const;
-		//! open discountFunction to public
-		DiscountFactor discount(const Array& x, Time t) const;
+        //! return whether there is a constraint at zero
+        bool constrainAtZero() const;
+        //! return weights being used
+        Array weights() const;
+        //! return l2 penalties being used
+        Array l2() const;
+        //! return optimization method being used
+        boost::shared_ptr<OptimizationMethod> optimizationMethod() const;
+        //! open discountFunction to public
+        DiscountFactor discount(const Array& x, Time t) const;
       protected:
         //! constructor
         FittingMethod(bool constrainAtZero = true, const Array& weights = Array(),
+                      const Array& l2 = Array(),
```

Comment:
To maintain backwards compatibility, the new parameter should go at the end of the argument list.
If you feel that the resulting order doesn't make sense, an alternative is to write another constructor and leave the existing one unchanged.

Reply:
The same goes for the other classes.

Reply:
Is there any reason not to do both?
Move it to the end of the existing constructor and add a new constructor?

Reply:
Well, you don't need two constructors taking the same arguments in different orders.